NewForce-Cohort-8/tabloidfullstack-suplex-syndicate
# Cb create post ticket6

**Description:**
Trello ticket 6: https://trello.com/c/LxzvXqYu/6-6-create-a-post

This ticket allows the logged in user to create a new post.

**Tasks done for this ticket:**
-  Post Model, Post Repository, IPost Repository, and Post Controller were updated on the server-side to accomplish a successful API POST request to create a post.
- PostForm.js was built and function with fetch call added to PostManager.js on the client-side.
- "/postform" route path added to Application Views as well as linked in the navbar on the client-side.

**Type of change**

**Testing Instructions**
1. git add and git commit whatever you're working on on your branch.
2. git checkout main.
3. git fetch -a.
4. git checkout my branch: cb-CreatePost-Ticket6.
5. If you don't see the most recent version of my files, git pull origin cb-CreatePost-Ticket6.
6.  (if not already done) Please open the TabloidFullStack.sln with Visual Studio 2022.
7.  (if not already done) **Install your dependencies** by running npm install from the same directory as your package.json fille.
8.  **launch the API** by clicking the execute with debugger solid green button in Visual Studio 2022.
12. Open a new GitBash window and CD into the client folder of the project and **open Visual Studio Code** with the command "code .", then launch the react app with the command "npm start".
13. In the react app, if you are not already logged in, you should be directed to the login page. Please **log in** with this email address: "foo@bar.com". This should redirect you back to the home page.
14. On nav bar, **click "New Post"**. You should be directed to a form to enter the Title, content, category (from drop-down selection), location of your image, and publish date time. Enter test data into each of these fields and click "Submit Post". Verify that you are redirected to the "/post/${post.id}" path for the details of the post, and your that post has been saved with all the information you entered. Please also verify that it displays the author display name for the user you are logged in as, as well as the publication date. 

Checklist:
